script/game.js

```diff
@@ -0,0 +1,250 @@
+        if (((weightUpperCard === ACE) && (takingField.className === "bordersFinalPack"))
+            || (((weightUpperCard - weightLowerCard) === 1)
+            && (removeCard.dataset.s === takingField.dataset.s))) {
+            replacementCard(takingField);
+            permissionToMove = true;
+        } else {
+            permissionToMove = false;
+        }
+
+        return permissionToMove;
+    }
+    
+    function replacementCard(takingField) {
+        var CARDS = 54;
+        var ACE = "1";
+        var nodeParent = takingField.parentNode;
+        gameModule.setCardsCounter();
+        if (cardCounter === CARDS) {
+            win.innerHTML = "YOU WIN";
+        }
+        
+        if (removeCard.dataset.n === ACE) {
+            takingField.appendChild(removeCard);
+        } else {
+            var n = +takingField.dataset.n + 1;
+            var s = +takingField.dataset.s;
+            var card = initialFieldModule.generateCard(n, s);
+            nodeParent.removeChild(takingField);
+            initialFieldModule.generateOpenCard(nodeParent, card);
+            nodeParent.appendChild(removeCard);
+        }
+        
+    }
+    
+    function checkTransferCard(step){
+        if (step !== 0) {
+            for (var i = 0; i < removedCardArr.length; i++) {
+                if (removedCardArr[i] === additionalCards[step - 1]) {
+                    checkTransferCard(step - 1);
+                    break;
+                } else if (lowerCard !== additionalCards[step - 1]) {
+                    lowerCard = additionalCards[step - 1];
+                    flagQueue = step - 1;
+                } 
+            }
+            initialFieldModule.removeChildren(div2);
+            initialFieldModule.generateOpenCard(div2, lowerCard);
+        } else if (div2.firstChild === null) {
+            div2.removeChild(div2.firstChild);
+        }
+    }
+    
+    return {
+        compareNumeric: function (a, b) {
+            if (a < b) {
+                
+                return 1;
+            }
+            
+            if (a > b) {
+                
+                return -1;
+            }
+        },
+        
+        rememberCard: function(card) {
+            gameModule.rememberParent(card);
+            removeCard = card;
+            
+            return card;
+        },
+        
+        resetAdditionalCards: function() {
+            additionalCards.length = 0;
+            transferCard = 0;
+        },
+        
+        rememberParent: function(card)
+        {
+            if (card.parentNode !== undefined) {
+                newOpenCard = card.parentNode;
+            }
+        }, 
+        
+        getAdditionalCards: function() {
+            
+            return additionalCards;
+        },
+        
+        rememberTransferCard: function(card, position) {
+            removedCardArr.push(card);
+            positionArr.push(position);
+        },
+        
+        openDownCard: function() {
```

Comment:
this function is quite big, probably it is possible to split it to subfunctions?